lib/openapi_first/definition/path_template.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module OpenapiFirst
+  # @visibility private
+  class PathTemplate
+    def initialize(template)
+      @template = template
+      @names = template.scan(NAMES_PAT).flatten
+      parts = template.split(CURLIES_PAT).map! do |part|
```

Comment:
Can we make this simpler / more efficient?

Reply:
I will take a look in the afternoon :)

Reply:
Not making it much faster, but I think we could replace the `part.sub(/{.*?}/, '([^/?#]+)')` with simpler `'([^/?#]+)'`. The effect of the `sub` is anyways `'([^/?#]+)'`, right? That's the `ALLOWED_PARAMETER_CHARACTERS` constant

all other changes are only my approach to readability, so feel free to ignore all of that or pick the parts you like :) I added a constant for the allowed parameter characters and the link to OAS spec so that's clear where is all the logic coming from. Apart from that I extracted the pattern building to a separate method

```ruby
module OpenapiFirst
  # @visibility private
  class PathTemplate
    # https://spec.openapis.org/oas/v3.1.0#path-templating
    TEMPLATE_EXPRESSION_REGEXP = /(\{[^}]+\})/
    TEMPLATE_EXPRESSION_NAME_REGEXP = /\{([^}]+)\}/
    ALLOWED_PARAMETER_CHARACTERS = '([^/?#]+)'

    def initialize(template)
      @template = template
      @names = template.scan(TEMPLATE_EXPRESSION_NAME_REGEXP).flatten
      @pattern = build_pattern(template)
    end

    def match(path)
      return {} if path == @template
      return if @names.empty?

      matches = path.match(@pattern)
      return unless matches

      values = matches.captures
      return if values.length != @names.length

      @names.zip(values).to_h
    end

    private

    def build_pattern(template)
      parts = template.split(TEMPLATE_EXPRESSION_REGEXP).map! do |part|
        part.start_with?('{') ? ALLOWED_PARAMETER_CHARACTERS : Regexp.escape(part)
      end

      %r{^#{parts.join}/?$}
    end
  end
end
```

I toyed around with the idea of having one method for returning names and parts, but it seemed to make it less readable and only removed one scan from the code which seemed like it's not worth the hassle

Reply:
I am out for today, but will take a look at this soon ™️ . Thanks for the suggestion 🤝

Reply:
>  The effect of the sub is anyways '([^/?#]+)', right?

Of course. Great find. Thanks.